src/encoder/external_encoder.ml

```diff
@@ -193,41 +200,49 @@ let encoder id ext =
 
   let encode h ratio frame start len =
     let channels = h.params.channels in
-    let start = Frame.audio_of_master start in
-    let b = AFrame.content_of_type ~channels frame start in
-    let len = Frame.audio_of_master len in
-    (* Resample if needed. *)
-    let b,start,len =
-      if ratio = 1. then
-        b,start,len
+    let sbuf =
+      if h.params.video then
+        Avi_encoder.encode_frame ~channels ~samplerate:h.params.samplerate ~converter:h.converter frame start len
```

Comment:
This line is too long. Too long for my terminal but also for the split diff view of github: one more reason to keep 80 char lines for readability!